### ISSUE TITLE
How to build openssl for ARM, ARM64 windows platforms?

### ISSUE THREAD

**Issue by bagajjal** [openssl/openssl#11289]
I would like to build openssl for ARM, ARM64 windows plaform. Is it supported?

The [build instructions](https://github.com/openssl/openssl/blob/master/INSTALL.md#windows) mentions only x86, x64. 

Visual studio doesn't have  Developer command prompt for ARM, ARM64. 
As per build instructions, we need to execute "nmake" from Developer command prompt. Since it's not there, how do we build?

If it supported, please provide the build instructions.

Thanks in advance.

**Comment by levitte** [comment]
For starters, you must make sure to have the right tools for the job.  This page should be able to answer most question on how to get what you need with Visual Studio:

https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019

We do have a few config targets for Windows that aren't mentioned in INSTALL.md yet...  they are fairly new and untested, but may be of interest to you.  They are:

VC-WIN32-ARM
VC-WIN64-ARM

If you want to build with UWP, here are the config targets to try:

VC-WIN32-ARM-UWP
VC-WIN64-ARM-UWP

Something to be aware of is that you will cross-compile, which means that the system you build on has nothing to do with the system you will run the programs and libraries on, so it's *highly* advisable to use `--prefix` when you configure.

You will also have to copy the resulting files manually to install them on the target system.  To make it easier on yourself, I suggest "installing" to a staging area of your choice (I use `S:\ome\where` as an example), then just copy the whole staging area where you want it:

```
nmake install DESTDIR=S:\ome\where
```

Please remember that you get to use things that we haven't tested much.  Also, please tell us about your experience.  Essentially, you get to help us testing these config targets.

**Comment by levitte** [comment]
I'm leaving this issue open for a while

**Comment by bagajjal** [comment]
@levitte - I am interested in VC-WIN32-ARM, VC-WIN64-ARM as they are using onecore.lib.
Is there a timeline by when these be fully tested and production ready?

**Comment by levitte** [comment]
Nope, or at least not yet. These are community provided config targets, so community testing and reporting back to us is the best way to help. You are part of the community, so I fully expect to hear back from you :wink:

**Comment by bagajjal** [comment]
Let me get this right. 

OpenSSL team (excluding community) hasn't tested OpenSSL on ARM32 and ARM64 windows platforms that uses onecore.lib?

**Comment by bagajjal** [comment]
Here is some context.

I'm from windows OpenSSH team. We are exploring the possible usage of OpenSSL as a crypto provider. To do so, we want OpenSSL to be production ready and build on all windows platforms (x86, x64, ARM, ARM64) using onecore.lib. 

If I understand correctly OpenSSL community has recently added support for ARM32 and ARM64 windows platforms and it's not fully tested i.e., not production ready.

Could you please confirm.

**Comment by richsalz** [comment]
What do *you* mean by production ready?  This is an open source project.  Do you want some kind of guarantee?  That won't happen.  You could offer pull requests of your own too.

**Comment by levitte** [comment]
So far we haven't.
Quite a number of our config targets are community provided, i.e. someone gave it away to us. That's exactly how we got the onecore.lib ones. They were tested by those who gave it to us at the time (which is fairly recently, I might add), and as I recall, I did some personal testing at the time to verify.
However, our regular testing with appveyor use the older VC-WIN32 and VC-WIN64A targets. 

What I hear from you is that it could be a good idea if we pushed forward and added the onecore.lib targets, as well or instead of what we currently test. Would that be satisfactory enough? Mind you, our possibility to do so will depend on the kind of support we can get from appveyor. I personally don't see a huge issue in doing this move.
